### PR TITLE
[dynamics] Add some template instantiation for integrator classes

### PR DESCRIPTION
**Pull request by ManifoldFR**
* base IntegratorAbstract
* base ExplicitIntegratorAbstract
* IntegratorRK2

### CODE REVIEW

**Comment by ManifoldFR** [comment]
@jorisv you can use your example project to check whether your executable or libaligator.so which contains the weak symbols for the integrators

File: include/aligator/modelling/dynamics/integrator-abstract.txx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/// @copyright Copyright (C) 2024 LAAS-CNRS, INRIA
+#pragma once
+
+#include "./integrator-abstract.hpp"
```
Comment (jorisv) [inline include/aligator/modelling/dynamics/integrator-abstract.txx:4]: On other txx file you're doing the following :

```cpp
#include "aligator/context.hpp"
#include "aligator/core/integrator-abstract.hpp"
```
I prefer the syntax with the absolute path and the context.hpp included before.

File: include/aligator/modelling/dynamics/integrator-explicit.txx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include "./integrator-explicit.hpp"
```
Comment (jorisv) [inline include/aligator/modelling/dynamics/integrator-explicit.txx:3]: Same than integrator-abstract.txx

File: include/aligator/modelling/dynamics/integrator-rk2.txx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/// @copyright Copyright (C) 2024 LAAS-CNRS, INRIA
+#pragma once
+
+#include "./integrator-rk2.hpp"
```
Comment (jorisv) [inline include/aligator/modelling/dynamics/integrator-rk2.txx:4]: Same than integrator-abstract.txx

File: include/aligator/modelling/dynamics/linear-ode.txx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+/// @copyright Copyright (C) 2024 LAAS-CNRS, INRIA
+#pragma once
+
+#include "./linear-ode.hpp"
```
Comment (jorisv) [inline include/aligator/modelling/dynamics/linear-ode.txx:4]: Same than integrator-abstract.txx

**Review by jorisv — CHANGES_REQUESTED** [review]

File: include/aligator/modelling/dynamics/integrator-abstract.txx (hunk 1 above)
Comment (ManifoldFR) [reply include/aligator/modelling/dynamics/integrator-abstract.txx:4]: Alright, will fix!

**Review by jorisv — APPROVED** [review]

**Pull request merged**